### ISSUE TITLE
BUG: Model::find() returns false instead of empty RecordSet

### ISSUE THREAD

**Issue by adamroyle** [UnionOfRAD/lithium#1365]
Add to `tests/integration/data/DatabaseTest.php` to see the bug in action.

```php
public function testNoResultsWithLimitAndHasMany() {
	$galleries = Galleries::find('all', [
		'conditions' => [
			'name' => 'not found'
		],
		'with' => 'Images',
		'limit' => 1, // comment out this line and the test passes
	]);
	$this->assertEqual([], $galleries->to('array'));
}
```

The bug occurs when you include a `hasMany` association combined with a `limit` and no results are returned.

This is due to some conditions inside `Database::read` and `Database::_queryExport`.

I don't think this is a new bug, but it does exist in 2.0 as I tested it against the latest dev branch.

**Comment by nateabele** [comment]
Would you be interested in pairing on it?

**Comment by adamroyle** [comment]
See my PR above with the test and fix. I also had to change a few things to get the test suite to run on php 8.2 on my local machine, but I didn't include those in this PR

**Comment by nateabele** [comment]
Fixed by #1368.
